server/src/mysql/mysql_writer.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::collections::HashMap;
+
+use common_types::datum::Datum;
+use opensrv_mysql::{Column, ColumnFlags, ColumnType, OkResponse, QueryResultWriter};
+
+use crate::handlers::sql::Response;
+
+pub struct MysqlQueryResultWriter<'a, W: std::io::Write> {
+    inner: Option<QueryResultWriter<'a, W>>,
+}
+
+impl<'a, W: std::io::Write> MysqlQueryResultWriter<'a, W> {
+    pub fn create(inner: QueryResultWriter<'a, W>) -> Self {
+        Self { inner: Some(inner) }
+    }
+
+    pub fn write(&mut self, query_result: Response) -> std::io::Result<()> {
+        if let Some(inner) = self.inner.take() {
+            return match query_result {
+                Response::AffectedRows(count) => Self::write_affected_rows(inner, count),
+                Response::Rows(rows) => Self::write_rows(inner, rows),
+            };
+        }
+        Ok(())
+    }
+
+    fn write_affected_rows(writer: QueryResultWriter<'a, W>, count: usize) -> std::io::Result<()> {
+        let res = OkResponse {
+            affected_rows: count as u64,
+            ..Default::default()
+        };
+        writer.completed(res)?;
+        Ok(())
+    }
+
+    fn write_rows(
+        writer: QueryResultWriter<'a, W>,
+        rows: Vec<HashMap<String, Datum>>,
+    ) -> std::io::Result<()> {
+        let default_response = OkResponse::default();
+        if rows.is_empty() {
+            writer.completed(default_response)?;
+            return Ok(());
+        }
+
+        fn convert_field_type(field: &Datum) -> ColumnType {
```

Comment:
Maybe we should define the `convert_field_type` and `make_column_by_field` outside the `write_rows`. Otherwise, it will be a little hard to write unit test for them.